patches/tModLoader/Terraria/ModLoader/ModResourceDisplaySet.cs

```diff
@@ -0,0 +1,573 @@
+
+			color = new Color(Main.mouseTextColor, Main.mouseTextColor, Main.mouseTextColor, Main.mouseTextColor);
+			if (ResourceOverlayLoader.PreDrawResourceDisplay(stats, this, false, ref color, out drawText))
+				DrawMana(Main.spriteBatch);
+			ResourceOverlayLoader.PostDrawResourceDisplay(stats, this, false, color, drawText);
+		}
+
+		/// <summary>
+		/// Allows you to initialize fields, textures, etc. before drawing occurs
+		/// </summary>
+		/// <param name="snapshot">A copy of <see cref="PlayerStats"/></param>
+		public virtual void PreDrawResources(PlayerStatsSnapshot snapshot) { }
+
+		/// <summary>
+		/// Draw the life resources for your display set here
+		/// </summary>
+		/// <param name="spriteBatch"></param>
+		public virtual void DrawLife(SpriteBatch spriteBatch) { }
+
+		/// <summary>
+		/// Draw the mana resources for your display set here
+		/// </summary>
+		/// <param name="spriteBatch"></param>
+		public virtual void DrawMana(SpriteBatch spriteBatch) { }
+
+		public void TryToHover() {
+			if (PreHover(out bool hoveringLife)) {
+				if (hoveringLife)
+					CommonResourceBarMethods.DrawLifeMouseOver();
+				else
+					CommonResourceBarMethods.DrawManaMouseOver();
+			}
+		}
+
+		/// <summary>
+		/// Allows you to specify if the vanilla life/mana hover text should display
+		/// </summary>
+		/// <param name="hoveringLife">Whether the hover text should be for life (<see langword="true"/>) or mana (<see langword="false"/>)</param>
+		public virtual bool PreHover(out bool hoveringLife) {
+			hoveringLife = false;
+			return false;
+		}
+
+		/// <summary>
+		/// Creates a drawing context for drawing a life resource from the Classic display set
+		/// </summary>
+		/// <param name="spriteBatch"></param>
+		/// <param name="heartNumber">Which life to draw, ranging from 1 to at most 20</param>
+		/// <param name="position">The position to draw the life at</param>
+		/// <param name="sourceFrame">The area within the life texture to draw.  Defaults to the entire texture</param>
+		/// <param name="color">The color to draw the life with.  Defaults to <see cref="Color.White"/></param>
+		/// <param name="rotation">The rotation to draw the life with.  Defaults to 0 radians</param>
+		/// <param name="origin">The relative center within the frame for rotation and scaling.  Defaults to the center of the frame</param>
+		/// <param name="scale">The scale to draw the life at.  Defaults to <see cref="Vector2.One"/></param>
+		/// <param name="pulse">Whether the life's scale should be modified by the "pulse" effect.  Defaults to <see langword="false"/></param>
+		/// <returns><see langword="null"/> if the parameters are invalid, otherwise the created context</returns>
+		public static ResourceOverlayDrawContext? GetClassicHeart(SpriteBatch spriteBatch, int heartNumber, Vector2 position, Rectangle? sourceFrame = null, Color? color = null, float rotation = 0, Vector2? origin = null, Vector2? scale = null, bool pulse = false) {
```

Comment:
These static helpers look really hard to maintain, and aren't used by `ExampleReversedBarsDisplay`